app/controllers/api/base_controller.rb

```diff
@@ -47,6 +47,10 @@ def controller_permission
       controller_name
     end
 
+    def current_permission
```

Comment:
Could you move the existing impl in application_controller.rb into FindCommon or maybe even ApplicationShared, since they're the same for now? It should avoid adding it again here.